From 629b83c413ad0c6dea83c9e6052e40fd75faf255 Mon Sep 17 00:00:00 2001
From: Finley Xiao <finley.xiao@rock-chips.com>
Date: Fri, 6 Jul 2018 17:11:50 +0800
Subject: [PATCH] soc: rockchip: opp_select: Parse 'rockchip,pvtm-thermal-zone'
 first for pvtm

Sometimes pvtm thermal zone is different from wide-temperature control
thermal zone, so let pvtm parse 'rockchip,pvtm-thermal-zone' first.

Change-Id: Ia6bdfbabb99b053864cbc3cbf14aac675e26483c
Signed-off-by: Finley Xiao <finley.xiao@rock-chips.com>
---
 .../devicetree/bindings/cpufreq/cpufreq-rockchip.txt          | 2 ++
 drivers/soc/rockchip/rockchip_opp_select.c                    | 4 ++--
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/Documentation/devicetree/bindings/cpufreq/cpufreq-rockchip.txt b/Documentation/devicetree/bindings/cpufreq/cpufreq-rockchip.txt
index 01b16ad55c43..617572aa2d15 100644
--- a/Documentation/devicetree/bindings/cpufreq/cpufreq-rockchip.txt
+++ b/Documentation/devicetree/bindings/cpufreq/cpufreq-rockchip.txt
@@ -47,6 +47,8 @@ In 'operating-points-v2' table:
                        used when current temperature is below reference
                        temperature. Conversely, The second one is used when
                        current temperature is above reference temperature.
+- rockchip,pvtm-thermal-zone: A thermal zone node containing thermal sensor,
+                       it's used to get the current temperature.
 - rockchip,thermal-zone: A thermal zone node containing thermal sensor,
                        it's used to get the current temperature.
 
diff --git a/drivers/soc/rockchip/rockchip_opp_select.c b/drivers/soc/rockchip/rockchip_opp_select.c
index 61c290ab4913..da9c538b2df8 100644
--- a/drivers/soc/rockchip/rockchip_opp_select.c
+++ b/drivers/soc/rockchip/rockchip_opp_select.c
@@ -232,9 +232,9 @@ static int rockchip_parse_pvtm_config(struct device_node *np,
 	if (of_property_read_u32_array(np, "rockchip,pvtm-temp-prop",
 				       pvtm->temp_prop, 2))
 		return -EINVAL;
-	if (of_property_read_string(np, "rockchip,thermal-zone",
+	if (of_property_read_string(np, "rockchip,pvtm-thermal-zone",
 				    &pvtm->tz_name)) {
-		if (of_property_read_string(np, "rockchip,pvtm-thermal-zone",
+		if (of_property_read_string(np, "rockchip,thermal-zone",
 					    &pvtm->tz_name))
 			return -EINVAL;
 	}
-- 
2.35.3

